xrpl/asyncio/wallet/wallet_generation.py

```diff
@@ -150,34 +154,40 @@ def process_faucet_host_url(input_url: str) -> str:
     return final_url
 
 
-def get_faucet_url(url: str, faucet_host: Optional[str] = None) -> str:
+def get_faucet_url(network_id: Optional[int], faucet_host: Optional[str] = None) -> str:
     """
-    Returns the URL of the faucet that should be used, based on whether the URL is from
-    a testnet or devnet client.
+    Returns the URL of the faucet that should be used, based on network_id and
+    faucet_host
 
     Args:
-        url: The URL that the client is using to access the ledger.
-        faucet_host: A custom host to use for funding a wallet.
+        network_id: The network_id corresponding to the XRPL network. This is parsed
+            from a ServerInfo() rippled response
+        faucet_host: A custom host to use for funding a wallet. This is useful because
+            network_id of sidechains might not be well-known. If faucet_url cannot be
+            gleaned from network_id alone, faucet_host is used
 
     Returns:
         The URL of the matching faucet.
 
     Raises:
-        XRPLFaucetException: if the provided URL is not for the testnet or devnet.
+        XRPLFaucetException: if the provided network_id does not correspond to testnet
+            or devnet.
     """
+    if network_id is not None:
```

Comment:
This check shouldn't be needed (or should throw an error) because the network ID should never be `None`.

Reply:
well, network_id is `None` in some cases -- the rippled integration tests do not specify a `network_id` in the config file, hence this value is `None`

I've updated the config file in one of the recent xrpl-py PRs to fix this issue. Nevertheless, I don't think it's good idea to throw an error if `network_id == None`